core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -148,10 +145,11 @@
     private final ParseFieldMatcher parseFieldMatcher;
 
     @Inject
-    public SearchService(Settings settings, ClusterService clusterService, IndicesService indicesService,IndicesWarmer indicesWarmer, ThreadPool threadPool,
+    public SearchService(Settings settings, SearchSettings searchSettings, ClusterService clusterService, IndicesService indicesService,IndicesWarmer indicesWarmer, ThreadPool threadPool,
```

Comment:
can we just fold `SearchSettings` into `SearchService` we already have enough classes and abstractions I don't think we need another one.